src/FSharpPlus/Extensions/Array.fs

```diff
@@ -25,6 +25,21 @@ module Array =
     let lift2 f x y =
         let lenx, leny = Array.length x, Array.length y
         Array.init (lenx * leny) (fun i -> f x.[i / leny] y.[i % leny])
+        
+        
+    /// <summary>Combines all values from three arrays and calls a mapping function on this combination.</summary>
+    /// <param name="f">Mapping function taking three element combination as input.</param>
+    /// <param name="x1">First array.</param>
+    /// <param name="x2">Second array.</param>
+    /// <param name="x3">Third array.</param>
+    ///
+    /// <returns>Array with values returned from mapping function.</returns>
+    let lift3 f x y z =
```

Comment:
Can we simplify to smth like this?
```fsharp
let lift3 (f: 'a->'b->'c->'d) (x: 'a[]) (y: 'b[]) (z: 'c[]) =
    Array.lift2 f x y 
    |> Array.lift2 (fun z' f' -> f' z' ) z 
```

Reply:
Similar approach could be applied to other `lift3`s

Reply:
Here again, I would check if there is no performance penalty before simplifying the code.

Reply:
I've measured the performance and it seems like it's faster than now, I'll fix it in next PR